### DISCUSSION TITLE
How to add delay in between write text  or commands.

### DISCUSSION THREAD

**Discussion by ClassickTech** [streamdeck-linux-gui/streamdeck-linux-gui#121] [Q&A]
Windows 11 user here who recently converted over to Linux Mint. Everything is working fine, except I don't know how to add delays between "Command" and "Write Text". I downloaded a program called xdotools but I got confused and wouldn't even know where to begin... or know if that's even the right tool in the first place. Any help? Also is there any documentation on how to get things done on this? I can't seem to find it. Thanks in advance.

Update: turns out Linux has a terminal command called "sleep". All good to go! 

2nd Update: First sleep command is not working well. lol

**Comment by Danie10** [comment]
xdotools is really powerful but I've also only scratched the surface and got very excited in my post at https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/56 about it. This tooks me a few days to figure out by trail and error. So just start small and build up.

From that, I see you need to use it in commands. But I'm not 100% sure we can actually mix commands and keypresses and text for example together in streamdeck-linux-gui. Normally I just use one of those, and in commands I'd build up the multiple commands in there. Maybe someone else must just come in and advise here.

So I'd try string my commands together with xdotool, and otherwise even get it to export any write text. There is some guidance at https://manpages.ubuntu.com/manpages/trusty/man1/xdotool.1.html which includes delay, as essentially those commands following the options, are all the chained commands.

**Reply by ClassickTech** [reply]
How exactly are you executing bash scripts with 1 button? So far I have tried:
Command: gnome-terminal
Write Text: bash media/john/12tb/scripts/master.sh

But the text output is coming out too fast and cutting the words off.

I have also tried:
Command: gnome-terminal -- bash media/john/12tb/scripts/master.sh

Nothing happens. 

Right now I just have to have a separate button to first open terminal. Then a 2nd button to execute the script. But I want to do it all in one press. 
Any help?

**Comment by Quintar** [comment]
To start a script just enter the FULL path into the command field. No need
to invoke the terminal. (But if you do you still need the full path like
/home/username/script.sh )
And yes you can do multiple things with one button press.
Also to pause between commands in a script just add pause <seconds>

Classick Tech ***@***.***> schrieb am Di., 21. Nov. 2023,
03:21:

**Comment by Danie10** [comment]
Seems there is more than one way as I also used the konsole command for KDE, but this is how I'm running some bash command line stuff in commands field:

`bash -c "obs-cli open 'Arch Linux Manjaro Tips'"`

**Comment by ClassickTech** [comment]
> To start a script just enter the FULL path into the command field. No need to invoke the terminal. (But if you do you still need the full path like /home/username/script.sh ) And yes you can do multiple things with one button press. Also to pause between commands in a script just add pause <seconds> Classick Tech ***@***.***> schrieb am Di., 21. Nov. 2023, 03:21:
> […](#)
> How exactly are you executing bash scripts with 1 button? So far I have tried: Command: gnome-terminal Write Text: bash media/john/12tb/scripts/master.sh But the text output is coming out too fast and cutting the words off. I have also tried: Command: gnome-terminal -- bash media/john/12tb/scripts/master.sh Nothing happens. Right now I just have to have a separate button to first open terminal. Then a 2nd button to execute the script. But I want to do it all in one press. Any help? — Reply to this email directly, view it on GitHub <[#121 (reply in thread)](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/121#discussioncomment-7625472)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABJEWWW6IGWIZEAOU6X6OWLYFQF2XAVCNFSM6AAAAAA7TGRXIOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMRVGQ3TE> . You are receiving this because you are subscribed to this thread.Message ID: <streamdeck-linux-gui/streamdeck-linux-gui/repo-discussions/121/comments/7625472 @github.com>


What am I doing wrong here:
Command: /12tb/scripts/master.sh 
Error: The command failed to execute.

Also tried:
Command: bash /12tb/scripts/master.sh 
Error: nothing happens.

**Reply by coolapso** [reply]
maybe this can help: https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/116#discussioncomment-7492882
